processor/spanmetricsprocessor/processor.go

```diff
@@ -379,7 +447,27 @@ func buildKey(serviceName string, span pdata.Span, optionalDims []Dimension) met
 		if attr, ok := spanAttr.Get(d.Name); ok {
 			value = tracetranslator.AttributeValueToString(attr)
 		}
-		concatDimensionValue(&metricKeyBuilder, value, true)
+		concatDimensionValue(&metricKeyBuilder, value)
+	}
+
+	k := metricKey(metricKeyBuilder.String())
+	return k
+}
+
+func buildResourceAttrKey(serviceName string, optionalResourceAttrs []Dimension, resourceAttr pdata.AttributeMap) metricKey {
```

Comment:
I'm thinking the `metricKey` is causing confusion. What is built here is serialised resource attributes keys. If we rename this type to `serializedKeys` instead will make more sense?

Reply:
I think something like `resourceAttrKey` might be better - imo we should mention resource attributes specifically since it's looking at those keys specifically not just any keys.

In that case we should / will have to make a second cache to store these, are we good with doing this? Or is there a better way to do it that I'm not seeing. (because the cache assumes the key is `metricKey`)

Reply:
yeah, we can call it `resourceKey`. I've no concern regarding using two caches.